packages/paste-website/src/components/tokens-list/token-card/index.tsx

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+import camelCase from 'lodash.camelcase';
+import {Box} from '@twilio-paste/box';
+import {Text} from '@twilio-paste/text';
+import {Button} from '@twilio-paste/button';
+import {Tooltip, useTooltipState} from '@twilio-paste/tooltip';
+import {useClipboard} from '@twilio-paste/clipboard-copy-library';
+import {ScreenReaderOnly} from '@twilio-paste/screen-reader-only';
+import {CopyIcon} from '@twilio-paste/icons/esm/CopyIcon';
+import {rgbToHex} from '../../../utils/rgbToHex';
+import {BackgroundColor} from './preview/BackgroundColor';
+
+interface TokenExampleProps {
+  category: string;
+  name: string;
+  value: string;
+}
+const PreviewComponent: React.FC<TokenExampleProps> = ({category, name, value}) => {
+  switch (category) {
+    case 'background-colors':
+      return <BackgroundColor name={name} value={value} />;
+    default:
+      return <Box>{value}</Box>;
+  }
+};
+
+export interface TokenCardProps {
+  category: string;
+  name: string;
+  value: string;
+  comment: string;
+}
+export const TokenCard: React.FC<TokenCardProps> = ({category, name, value, comment}) => {
+  const tooltipState = useTooltipState();
+  const [tooltipText, setTooltipText] = React.useState('Copy token name');
+  // Prevents tooltip being visible on first render due to reakit positioning bug code
+  const isFirstRender = React.useRef(true);
+  const clipboard = useClipboard({copiedTimeout: 2000});
+  const camelCaseName = camelCase(name.replace('$', ''));
+
+  const handleCopyName = React.useCallback(() => {
+    clipboard.copy(camelCaseName);
+  }, [camelCaseName]);
+
+  React.useEffect(() => {
+    setTooltipText(clipboard.copied ? 'Copied!' : 'Copy token name');
+  }, [clipboard.copied]);
+
+  // Reakit has a bug where the tooltip doesn't recalc position on content changes
+  // This is a workaround until we upgrade to Ariakit with Floating UI fixes
+  // https://github.com/twilio-labs/paste/discussions/2037
+  React.useEffect(() => {
+    // This prevents the tooltip from showing up on mount
+    if (isFirstRender.current) {
+      isFirstRender.current = false;
```

Comment:
thought(outloud): I really need to read a "what you can use refs for" blog post because I would never have thought to have used a ref for this.

Reply:
I wouldn't worry about these use-cases too much. This is one of those "I didn't know I could do this but I figured it out because I needed to" situations.